app/assets/javascripts/live_search.js

```diff
@@ -134,50 +160,63 @@
   }
 
   LiveSearch.prototype.formChange = function formChange (e) {
-    var pageUpdated
     if (this.isNewState()) {
-      this.getTaxonomyFacet().update()
+      this.getAndUpdateTaxonomyFacet()
       this.saveState()
       this.updateOrder()
       this.updateLinks()
       this.updateTitle()
-      pageUpdated = this.updateResults()
-      pageUpdated.done(
-        function () {
-          var newPath = window.location.pathname + '?' + $.param(this.state)
-          window.history.pushState(this.state, '', newPath)
-          this.trackingInit()
-          this.setRelevantResultCustomDimension()
-          this.trackPageView()
-        }.bind(this)
-      )
+      this.updateResults()
+    }
+  }
+
+  LiveSearch.prototype.serializeState = function (state) {
+    var params = []
+    if (Array.isArray(state)) {
+      for (var i = 0; i < state.length; i++) {
+        params.push(state[i].name + '=' + state[i].value)
+      }
+    } else {
+      for (var key in state) {
+        if (Object.prototype.hasOwnProperty.call(state, key)) {
+          params.push(encodeURIComponent(key) + '=' + encodeURIComponent(state[key]))
+        }
+      }
     }
+    return params.join('&')
   }
```

Comment:
I'm generally against polyfills because they can add more code than needed (or known) and once added, they need to be maintained and probably removed further down the road. Writing it this way is a bit delicate but I think clearer (and I learned plenty about form serialisation in the process!).

If we did more form handling like this then I'd definitely want a way of modernising and reusing the code better, but as the search code is a weird standalone anomaly I think it's okay.